aptoide-database/src/main/java/cm/aptoide/pt/database/room/NotificationDao.java

```diff
@@ -0,0 +1,36 @@
+package cm.aptoide.pt.database.room;
+
+import androidx.room.Dao;
+import androidx.room.Insert;
+import androidx.room.Query;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import java.util.List;
+
+import static androidx.room.OnConflictStrategy.REPLACE;
+
+@Dao public interface NotificationDao {
+
+  @Query("SELECT * from NOTIFICATION where type IN (:notificationType) AND dismissed BETWEEN :startingTime AND :endTime ")
+  Single<List<RoomNotification>> getDismissed(Integer[] notificationType, long startingTime,
+      long endTime);
+
+  @Query("SELECT * from NOTIFICATION where type IN (:notificationType) ORDER BY timeStamp DESC")
+  Single<List<RoomNotification>> getAllSortedDescByType(Integer[] notificationType);
```

Comment:
Why Integer[], seems like we always pass a single int.

Reply:
In this case, the notification type does not come from the SocialPolicy, but it comes from the mapping of the notification id  (getNotificationType method on NotificationIdsMapper), a method thast returns an array. Eventhough for some cases it returns only one notification type, there are others where it returns more than one (case 1: line 36).